usr.bin/ssh/sshd.c

```diff
@@ -725,14 +720,16 @@ pack_config(struct sshbuf *conf)
 }
 
 static void
-send_rexec_state(int fd, struct sshbuf *m)
+send_rexec_state(int fd, struct sshbuf *config)
 {
+	struct sshbuf *m;
 	u_int mlen;
 	int sz;
 
-	debug3_f("entering fd = %d rexec_state len %zu", fd,
-	    sshbuf_len(m));
+	debug3_f("entering fd = %d config len %zu", fd,
+	    sshbuf_len(config));
 
+	m = pack_config(config);
```

Comment:
We could pack the config once and only serialise/send the keys after the config blob has been sent. E.g. 

1. Make a global `sshbuf *config_blob`
2. Fill `child->config` using `sshbuf_fromb(config_blob)`, so it's zero-copy
3. Add a new `child->keys` that only gets filled in and sent once `child->config` has finished sending

The benefitd to this are 1) less work for the listener it doesn't have to serialise the config each time, 2) less memory as it doesn't have to hold on to `max_startup * sshbuf_len(child->config)` in the worst case, and 3) the keys are exposed in memory for much less time (I expect that in the usual case they would instantly fit in the socket buffer).

OTOH it's a fair bit of work